types/wlr_output.c

```diff
@@ -168,6 +170,28 @@ bool wlr_output_set_cursor(struct wlr_output *output,
 				WL_SHM_FORMAT_ARGB8888, stride, width, height, buf);
 }
 
+bool wlr_output_set_cursor_surface(struct wlr_output *output,
+		struct wlr_surface *surface, int32_t hotspot_x, int32_t hotspot_y) {
+	if (output->impl->set_cursor) {
+		output->impl->set_cursor(output, NULL, 0, 0, 0, 0, 0);
+		wlr_log(L_INFO, "TODO: wlr_output_set_cursor_surface for hw cursors");
+	}
+
+	output->cursor.is_sw = true;
+	output->cursor.width = surface->current->width;
+	output->cursor.height = surface->current->height;
+	output->cursor.hotspot_x = hotspot_x;
+	output->cursor.hotspot_y = hotspot_y;
+
+	wlr_texture_destroy(output->cursor.texture);
+	output->cursor.texture = surface->texture;
+
+	wlr_renderer_destroy(output->cursor.renderer);
+	output->cursor.renderer = surface->renderer;
+
+	return true;
```

Comment:
This outputs `attempt to render invalid texture`, I don't understand why.

Reply:
This is a return statement. It doesn't output anything.

Reply:
Yeah, that's exactly why I don't understand why it outputs something.

(I was talking about the entire function)

Reply:
Well what line does the output actually *happen* on?

Reply:
So actually this isn't the place where the cursor is rendered on screen. This just sets `output->cursor`, the cursor is rendered in `wlr_output_swap_buffers`.

Reply:
Aye. And it's there where the error occurs?

Reply:
Yeah, the error occurs when calling `wlr_render_with_matrix` from `wlr_output_swap_buffers`. I don't get why.

Reply:
It means the texture hasn't had pixels attached to it yet.

Reply:
Hmm, thanks.